lib/wasi/src/state/mod.rs

```diff
@@ -376,6 +376,7 @@ impl WasiFs {
         }
     }
 
+    /// Returns the next available inode index for creating a new inode.
```

Comment:
As written, someone might think this can return the same value twice. This returns a new inode number, or allocates an inode, or something like that?

Reply:
I thought that too, but it takes a `&mut self` which I thought makes it sufficiently clear that it's not a pure function